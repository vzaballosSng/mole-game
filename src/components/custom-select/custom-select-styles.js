import { css } from 'lit';

export const customSelectStyles = css`
  label {
    margin-right: .9375rem;
  }

  select {
    padding: .625rem 0;
    border: none;
    border-bottom: 2px solid white;
    color: white;
    font-size: 1rem;
    outline: none;
    background: transparent;
  } 
`;
