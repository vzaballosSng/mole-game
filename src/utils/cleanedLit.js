// eslint-disable-next-line no-undef
const { prettyDOM } = require('@testing-library/dom');

export const cleanedLit = (htmlElement) => {
	const domHTML = prettyDOM(htmlElement, Infinity);
	const cleaned = domHTML.replace(/"<!--\?lit\$.*?-->"/g, '""');
	return cleaned;
};
export const cleanedLitElements = (htmlElements) => {
	const cleaned = htmlElements.map((htmlElement) => {
		return cleanedLit(htmlElement);
	});
	return cleaned;
};
