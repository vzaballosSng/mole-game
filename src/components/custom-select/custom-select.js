import { LitElement, html } from 'lit';
import { customSelectStyles } from './custom-select-styles.js';

export class CustomSelect extends LitElement {
	static get properties() {
		return {
			listOptions: { type: Array },
      labelName: { type: String, attribute: 'label-name' }
		};
	}

	static get styles() {
		return [customSelectStyles];
	}

	constructor() {
		super();
    this.labelName = '';
    this.listOptions = [];
	}

  get _renderOptions() {
    return this.listOptions.map(option => {
      return html` <option value="${option}">${option}</option>`;
    })
  }

  _sendSelectValue(e) {
    this.dispatchEvent(new CustomEvent('change-select-value', {
      detail: {
        value: e.target.value,
      },
      bubbles: true,
			composed: true,
    }));
  }


	render() {
		return html`
      <label for="level">${this.labelName}</label>
			<select
        aria-labelledby="level"
        aria-label="level"
        id="level"
        name="select-level"
        @change=${this._sendSelectValue}
      >
        ${this._renderOptions}
			</select>
		`;
	}
}

window.customElements.define('custom-select', CustomSelect);
