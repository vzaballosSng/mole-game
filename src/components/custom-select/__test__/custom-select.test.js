import { html, fixture } from '@open-wc/testing';
import { expect } from '@jest/globals';
import { axe, toHaveNoViolations } from 'jest-axe';
import { cleanedLit } from '../../../utils/cleanedLit.js';
import '../custom-select.js';

expect.extend(toHaveNoViolations);

describe('custom-select', () => {
	it('custom-select matches the snapshot', async () => {
		const el = await fixture(
			html`<custom-select></custom-select>
			>`
		);
		const dataCleaned = cleanedLit(el);
		expect(dataCleaned).toMatchSnapshot();
	});
	
	it('accessibility test', async () => {
		const el = await fixture(html`<custom-select></custom-select>`);
		const results = await axe(el);
		expect(results).toHaveNoViolations();
	});
});
