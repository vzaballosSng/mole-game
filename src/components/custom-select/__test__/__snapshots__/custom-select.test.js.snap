// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`custom-select custom-select matches the snapshot 1`] = `"[36m<custom-select />[39m"`;
